### ISSUE TITLE
FB13683971: Add the ability to get the user's presented desktop background for the current screen and space

### ISSUE THREAD

**Issue by sindresorhus** [feedback-assistant/reports#478]
- **Date:** 2024-03-12
- **Resolution:** Open
- **Area:** AppKit
- **OS:** macOS 14.4
- **Type:** Suggestion

## Description

Before I continue, this is not the same as `NSWorkspace.shared.desktopImageURL`.

Some apps want to use the user’s desktop background in some contexts. For example, Mission Control displays a blurred version of the users desktop background. It would be useful if third-party apps could easily use this effect too.

My personal use-case is that I develop a calendar app and it shows fullscreen notifications for when events start. In this fullscreen window, I display a blurred version of the user’s desktop background. Currently, I use `CGWindowListCopyWindowInfo` to get the user’s desktop background for this. But that API is now deprecated, which leaves me with no alternatives. Some other use-cases include showing a break reminder.

Why can I not use `NSWorkspace.shared.desktopImageURL`? In short, I need the exact background shown on the current screen and space, not the wallpaper the user has configured. The `desktopImageURL` method has a lot of limitations and bugs:
- If a folder is specified in System Settings for the wallpaper, a folder URL is returned here too, which makes it impossible to know which exact image that is shown.
- It does not support dynamic wallpapers.
- It does not support spaces.
- It sometimes returns `nil` for no reason.

I’m also aware there is ScreenCaptureKit, but it doesn’t make sense to force the user to give my app screen recording permission just to get the desktop background.

Some potential solutions:
1. NSWorkspace.shared.presentedDesktopBackground(screen: .main) - which would return exactly what’s shown on the desktop (if the wallpaper is cropped, it would be cropped too)
2. SwiftUI ShapeStyle: `view.background(.desktopBackground)`
3. SwiftUI Material: `view.background(.desktopBackground)` - this would be an already blurred version of the desktop background, which is all I need.

tl;dr; I need a screenshot of the desktop background window (it’s owned by Dock.app).
